x-pack/plugins/maps/public/classes/layers/blended_vector_layer/blended_vector_layer.ts

```diff
@@ -187,14 +190,10 @@ export class BlendedVectorLayer extends VectorLayer implements IVectorLayer {
     this._clusterStyle = new VectorStyle(clusterStyleDescriptor, this._clusterSource, this);
 
     let isClustered = false;
-    const sourceDataRequest = this.getSourceDataRequest();
-    if (sourceDataRequest) {
-      const requestMeta = sourceDataRequest.getMeta();
-      if (
-        requestMeta &&
-        requestMeta.sourceMeta &&
-        requestMeta.sourceMeta.sourceType === SOURCE_TYPES.ES_GEO_GRID
-      ) {
+    const countDataRequest = this.getDataRequest(ACTIVE_COUNT_DATA_ID);
+    if (countDataRequest) {
+      const requestData = countDataRequest.getData() as CountData;
```

Comment:
Consider adding this same `as CountData` cast here https://github.com/elastic/kibana/blob/539cfa9a8b9949e5ddc8acdd2eace4cf59f30c4c/x-pack/plugins/maps/public/classes/layers/blended_vector_layer/blended_vector_layer.ts#L286 for clarity.